### DISCUSSION TITLE
Query Service support in sdk is really hard to build helper around

### DISCUSSION THREAD

**Discussion by zveznicht** [ydb-platform/ydb-go-sdk#1568] [Q&A] [answered]
Query Service [interface](https://github.com/ydb-platform/ydb-go-sdk/blob/a63ea4f5feb3029c81ade63f71ffc8a0598f80b8/query/client.go) is full of internal packages, which are not accessible from outside. Most importantly `options` package content. I guess it is intentional to make users use builders for them. But that makes it really hard to use them in client's code.

For example we have this case: 
We have a configuration to define DB path (directory) where all tables (and other objects) live. And from the code point of view this is just part of a YDB client's configuration: the code that uses for example `table.Client` simply makes query as if tables were in the DB root. 
To achieve this we have a `table.Client` wrapper (it implements ydb sdk `table.Client`) which adds `PRAGMA TablePathPrefix` to every query. This is very handy and (most importantly) allows other code to work both with pure sdk `table.Client` and our wrapper. For `query.Service` this is obviously impossible, because it is full of internal types. 

Even if you don't want to expose your internal types, you still can make  type aliases for them inside query package. Clients still won't be able to properly build their own implementation, but they can at least reference them.

**Comment by qrort** [comment, marked as answer]
The main problem of the Query Service interface was internal `options.Execute`. There was public alias made for it in https://github.com/ydb-platform/ydb-go-sdk/commit/9d15bdf15219d1fe6a716780c263d611afe42332, and as of now, there are no internal types in interfaces.
